heron/spi/src/java/com/twitter/heron/spi/statemgr/SchedulerStateManagerAdaptor.java

```diff
@@ -179,6 +189,60 @@ public Boolean deleteSchedulerLocation(String topologyName) {
     return awaitResult(delegate.getSchedulerLocation(null, topologyName));
   }
 
+  /**
+   * Checks to see if the execution state exists
+   *
+   * @return a boolean indicating whether the execution state exists or not
+   */
+  public boolean doesExecutionStateExist(String topologyName) {
+    return awaitResult(delegate.executionStateExists(topologyName));
+  }
+
+  /**
+   * Clean all states of a heron topology
+   * 1. Topology def and ExecutionState are required to exist to delete
+   * 2. TMasterLocation, SchedulerLocation and PhysicalPlan may not exist to delete
+   */
+  public Boolean cleanState(String topologyName) {
```

Comment:
Not sure whether to put `cleanState(...)` here or in the Utils class since it is more like an utils method, providing an aggregation of primitive methods.